cpp/src/arrow/util/decimal.cc

```diff
@@ -261,19 +272,12 @@ bool ParseDecimalComponents(const char* s, size_t size, DecimalComponents* out)
   // Optional exponent
   if (StartsExponent(s[pos])) {
     ++pos;
-    if (pos == size) {
-      return false;
-    }
-    // Optional exponent sign
-    if (IsSign(s[pos])) {
-      out->exponent_sign = std::string(s + pos, 1);
+    if (pos != size && s[pos] == '+') {
       ++pos;
     }
-    pos = ParseDigitsRun(s, pos, size, &out->exponent_digits);
-    if (out->exponent_digits.empty()) {
-      // Need some exponent digits
-      return false;
-    }
+    out->has_exponent = true;
+    static internal::StringConverter<Int32Type> exponent_converter;
```

Comment:
Making this `static` relies on the converter being stateless (otherwise this is not thread-safe). Do you lose performance by removing the `static`?

Reply:
I will measure.  I did this because it was recommended by the converter class docs mentioned they could be expensive to instantiate (and seemed like they should be thread-safe).

Reply:
Looks like there is no performance impact (might have even improved it).